libraries/AP_InertialSensor/AP_InertialSensor_MPU9250.cpp

```diff
@@ -511,6 +625,50 @@ void AP_InertialSensor_MPU9250::_read_data_transaction()
 }
 
 /*
+  read data from FIFO and update filtered data
+ */
+void AP_InertialSensor_MPU9250::_read_fifo_data_transaction()
+{
+    uint8_t n_bytes, n_samples;
+    uint8_t rx[MPU9250_FIFO_MAX_SIZE];
+    int i = 0;
+
+    Vector3f accel, gyro;
+    Vector3f data_accel[MPU9250_FIFO_MAX_VECTOR3F_SIZE],
```

Comment:
why declare these arrays of size 85 here? can't we declare a single vector inside the loop?